hermes-common/src/main/java/pl/allegro/tech/hermes/common/metric/executor/InstrumentedExecutorServiceFactory.java

```diff
@@ -33,24 +33,51 @@ public ExecutorService getExecutorService(String name, int size, boolean monitor
         return monitoringEnabled ? monitor(name, executor) : executor;
     }
 
-    public ScheduledExecutorService getScheduledExecutorService(
-            String name, int size, boolean monitoringEnabled, boolean removeOnCancel
-    ) {
 
-        ThreadFactory threadFactory = new ThreadFactoryBuilder().setNameFormat(name + "-scheduled-executor-%d").build();
-        ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(size, threadFactory);
-        executor.setRemoveOnCancelPolicy(removeOnCancel);
-        return monitoringEnabled ? monitor(name, executor) : executor;
+    public class ScheduledExecutorServiceBuilder {
```

Comment:
I was thinking about overloaded constructor, but builder seems to be even better